install/operator/pkg/syndesis/backup/backup.go

```diff
@@ -193,35 +218,273 @@ func (b *Backup) ensureDir() (err error) {
+ * remote command.
+ * Cannot use 1 container since the termination state change is what
+ * is detected for triggering the execution of the remote command
+ */
 // Create a database backup
 func (b *Backup) backupDatabase() error {
-	api, err := b.apiClient()
+
+	b.logger("backup").Info("Initiating database backup ...")
+
+	// Load configuration to to use as context for generator pkg
+	sc, err := configuration.GetProperties(configuration.TemplateConfig, b.Context, b.client(), b.Syndesis)
+	if err != nil {
+		return err
+	}
+	b.synConfig = sc
+
+	dbURL, err := url.Parse(b.synConfig.Syndesis.Components.Database.URL)
 	if err != nil {
 		return err
 	}
 
-	pod, err := util.GetPodWithLabelSelector(api, b.Namespace, "syndesis.io/component=syndesis-db")
+	b.synConfig.Syndesis.Components.Database.BackupConfig = configuration.BackupConfiguration{
+		Host:     dbURL.Hostname(),
+		Port:     dbURL.Port(),
+		FileDir:  "/pgdata/syndesis-db-backups/*",
+		FileName: SYNDESIS_DUMP_FILENAME,
+	}
+
+	// Get migration resources, this should be the db migration job
+	resources, err := generator.Render("./backup/syndesis-backup-job.yml.tmpl", b.synConfig)
 	if err != nil {
 		return err
 	}
-	backupfile, err := os.Create(filepath.Join(b.BackupDir, b.backupPath, "syndesis-db.dump"))
+
+	// Install the resources
+	for _, res := range resources {
+		//
+		// The syndesis CR owns the job. This is necessary
+		// since the deploying of a job requires an OWnerReference
+		// complete with Name, UID & Kind.
+		//
+		operation.SetNamespaceAndOwnerReference(res, b.Syndesis)
+		_, _, err := util.CreateOrUpdate(b.Context, b.client(), &res)
+		if err != nil {
+			return err
+		}
+	}
+
+	return b.executeJob(BACKUP_JOB_NAME, b.bkpPodTask)
+}
+
+//
+// Execute a task within a Job
+//
+func (b *Backup) executeJob(jobName string, jobTask jobTaskFunc) error {
+	//
+	// TODO
+	//
+	// Since the job keeps dying after 8-12 seconds regardless of the pod
+	// still running (maybe a 3.11 issue?), it is necessary to monitor the
+	// target pod instead. Is the job is more trouble than its worth??
+	//
+	j := &batchv1.Job{}
+	if err := b.client().Get(b.Context, types.NamespacedName{Namespace: b.Namespace, Name: jobName}, j); err != nil {
+		return err
+	}
+
+	//
+	// Wait for the job to construct the pod
+	//
+	err := wait.Poll(POLL_INTERVAL, POLL_TIMEOUT, func() (done bool, err error) {
+		podList, err := util.PodsFromJob(b.Context, b.client(), j, b.Namespace)
+		if err != nil {
+			return false, err
+		}
+
+		if len(podList.Items) == 0 {
+			return false, nil
+		}
+
+		for _, myPod := range podList.Items {
+			//
+			// Found the pod so execute the pod task
+			//
+			err = wait.Poll(POLL_INTERVAL, POLL_TIMEOUT, jobTask(&myPod))
+			if err != nil {
+				return false, err
+			}
+
+			//
+			// Backup pod task completed successfully so job done!
+			//
+			return true, nil
+		}
+
+		//
+		// Still waiting on pods being created by job
+		//
+		return false, nil
+	})
+
```

Comment:
The wait into a wait approach is confusing. Can't we really use the job to figure out these stuff? If we can't use the job for this case, then it is probably better to spin up a pod directly because using a job that 'keeps dying' turns from being handy to being a trouble.